cpp/include/cudf/detail/sizes_to_offsets_iterator.cuh

```diff
@@ -269,5 +273,55 @@ auto sizes_to_offsets(SizesIterator begin,
   return last_element.value(stream);
 }
 
+/**
```

Comment:
The `make_offsets_child_column` is moved from `cudf::strings::detail` in `strings_children.cuh` to `cudf::detail` in this file. This utility is used for strings and lists functions.